clarity/contracts/key-token/key-usda-wbtc.clar

```diff
@@ -0,0 +1,152 @@
+(impl-trait .trait-ownable.ownable-trait)
+(impl-trait .trait-semi-fungible-token.semi-fungible-token-trait)
+
+(define-constant ERR-NOT-AUTHORIZED (err u1000))
+(define-constant ERR-TOO-MANY-POOLS (err u2004))
+(define-constant ERR-INVALID-BALANCE (err u2008))
+
+(define-fungible-token key-usda-wbtc)
+(define-map token-balances {token-id: uint, owner: principal} uint)
+(define-map token-supplies uint uint)
+(define-map token-owned principal (list 2000 uint))
+
+(define-data-var contract-owner principal tx-sender)
+(define-map approved-contracts principal bool)
+
+(define-read-only (get-owner)
+  (ok (var-get contract-owner))
+)
+
+(define-public (set-owner (owner principal))
+  (begin
+    (asserts! (is-eq contract-caller (var-get contract-owner)) ERR-NOT-AUTHORIZED)
+    (ok (var-set contract-owner owner))
+  )
+)
+
+(define-private (check-is-approved (sender principal))
+  (ok (asserts! (or (default-to false (map-get? approved-contracts sender)) (is-eq sender (var-get contract-owner))) ERR-NOT-AUTHORIZED))
+)
+
+(define-read-only (get-token-owned (owner principal))
+    (default-to (list) (map-get? token-owned owner))
+)
+
+(define-private (set-balance (token-id uint) (balance uint) (owner principal))
+    (begin
+	    (map-set token-balances {token-id: token-id, owner: owner} balance)
+        (map-set token-owned owner (unwrap! (as-max-len? (append (get-token-owned owner) token-id) u2000) ERR-TOO-MANY-POOLS))
+        (ok true)
+    )
+)
+
+(define-private (get-balance-or-default (token-id uint) (who principal))
+	(default-to u0 (map-get? token-balances {token-id: token-id, owner: who}))
+)
+
+(define-read-only (get-balance (token-id uint) (who principal))
+	(ok (get-balance-or-default token-id who))
+)
+
+(define-read-only (get-overall-balance (who principal))
+	(ok (ft-get-balance key-usda-wbtc who))
+)
+
+(define-read-only (get-total-supply (token-id uint))
+	(ok (default-to u0 (map-get? token-supplies token-id)))
+)
+
+(define-read-only (get-overall-supply)
+	(ok (ft-get-supply key-usda-wbtc))
+)
+
+(define-read-only (get-decimals)
+  	(ok u6)
+)
+
+(define-read-only (get-token-uri (token-id uint))
+	(ok none)
+)
+
+(define-public (transfer (token-id uint) (amount uint) (sender principal) (recipient principal) (memo (optional (buff 34))))
```

Comment:
The SFT trait does not have an optional memo as a function parameter. https://github.com/MarvinJanssen/sips/blob/feat/sip-013/sips/sip-013/sip-013-semi-fungible-token-standard.md#trait-definition